### ISSUE TITLE
Better support for Confluent's kafka AvroSerializer/AvroDeserializer

### ISSUE THREAD

**Issue by robbertvanwaveren** [marcosschroh/dataclasses-avroschema#405]
**Is your feature request related to a problem? Please describe.**

The confluent AvroSerializer/AvroDeserializer classes require both a `schema` as well as an object `to_dict` and `from_dict` to object function. Using the following code I can successfully use the `avro_schema`, `asdict` and `parse_obj` functions from the generated code from this library to fill those requirements: 

```
    avro_serializer = AvroSerializer(registry_client, msg.avro_schema(), lambda obj, sc: obj.asdict(standardize_factory=standardize_custom_type))
    avro_deserializer = AvroDeserializer(registry_client, msg.avro_schema(), lambda dict, sc: Message.parse_obj(dict))
```
However its still a bit clunky as the functions do not exactly match. Espescially the current implementation of the `to_dict` is close but not exactly usable as it does not fill the standardize_factory as above.

Note that successful integration is key to usage of these models in combination with an enterprise setting that relies of a schema-registry and confluents avro encoding header to support schema evolution across different implementation languages.

**Describe the solution you'd like**

I'd like the generated classes to come with class-level methods matching the provided lambdas, or if that is not feasible at least include the `standardize_factory=standardize_custom_type)` in the call to `asdict` from the object's to_dict by default.

**Describe alternatives you've considered**
-

**Additional context**
-

**Comment by marcosschroh** [comment]
@robbertvanwaveren 

> Note that successful integration is key to usage of these models in combination with an enterprise setting that relies of a schema-registry and confluents avro encoding header to support schema evolution across different implementation languages.

I agree but in this case the confluent python ecosystem is not stable at all so I would not go to far in the integration. However, I think we can include `standardize_factory: Optional[Callable[..., Any] = None` in the `to_dict` method . PR is welcome.

PS: Documentation and examples would be also nice to have.

**Comment by marcosschroh** [comment]
Another solution for this could be to have a `mixings.Confluent` that could add extra functionality for this use case

**Comment by leonardotorresaltez** [comment]
@marcosschroh , why do you think  confluent python ecosystem is not stable at all ?

**Comment by marcosschroh** [comment]
A few years ago I was using it and it was going well but out of the sudden they have changed the APIs a lot. If I recall correctly now topics, subjects and schemas are tied together and you have to use `from confluent_kafka.serialization import  SerializationContext` to produce an event, which it seems unnecessary from my point of view. [Example](https://github.com/confluentinc/confluent-kafka-python/blob/master/examples/avro_producer.py#L139-L142)

Maybe I am wrong but the changes that I have seen they do not make sense. I think going properly Open Source makes things really simple and you are not tied to a platform.

I use `aiokafka`, `dataclasses-avroschemas` and our microframework [kstreams](https://github.com/kpn/kstreams)
